legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-bigquery/src/main/java/org/finos/legend/engine/persistence/components/relational/bigquery/executor/BigQueryExecutor.java

```diff
@@ -70,6 +71,22 @@ public void executePhysicalPlan(SqlPlan physicalPlan, Map<String, String> placeh
         }
     }
 
+    public Map<StatisticName, Object> executeLoadPhysicalPlanAndGetStats(SqlPlan physicalPlan, Map<String, String> placeholderKeyValues)
+    {
+        List<String> sqlList = physicalPlan.getSqlList();
+
+        // The first SQL is a load statement
+        // Executed in a new transaction
+        Map<StatisticName, Object> loadStats = bigQueryHelper.executeLoadStatement(getEnrichedSql(placeholderKeyValues, sqlList.get(0)));
+
+        // The second SQL is an insert statement
+        // We need to first close the current transaction (if it exists) and open a new transaction
+        // Such that the result of the Load will be available to the Insert
+        bigQueryHelper.close();
```

Comment:
Not sure I follow.  This seems as some encapsulation is broken.  Like, where is the transaction created?  Should not the initiator of the transaction be the one closing it?

Reply:
Yeah i agree, I think it was done because INSERT had to be run in a new transaction. (As isolation level in Bigquery is Snapshot). So we were closing the current transaction and running INSERT a new transaction.

Changed the code as you have suggested. Not closing the current transaction and letting the initiator to close it.
Running INSERT in a new transaction.
Thanks for the suggestion!